src/components/Modal/index.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+    memo,
+    useRef,
+    useState,
+    forwardRef,
+    useImperativeHandle,
+} from 'react';
+
+import { useOutsideClick } from '../../hooks';
+import { Close as CloseIcon } from '../../icons';
+import { Button } from '../Button/Button';
+import Styled from './styles';
+import type { ModalProps, ModalBaseProps } from './types';
```

Comment:
I like this, since it is a lot of code, and might be imported by other files, this makes sense to separate, nice organization for components.